File: ELEC881_Tabnine/src/No Description Prompt/Python/q13-integer-replacement.py
class Solution(object):
    def integerReplacement(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        while n > 1:
            if n % 2 == 0:
                n //= 2
            else:
                if (n + 1) % 2 == 0 or (n - 1) % 2 == 0:
                    n -= 1
                else:
                    n += 1
            count += 1
        return count
        

File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q13-integer-replacement.js
/**
 * @param {number} n
 * @return {number}
 */
var integerReplacement = function(n) {
    let count = 0;
    while (n > 1) {
        if (n % 2 === 0) {
            n /= 2;
        } else if ((n + 1) % 4 === 0 || (n - 1) % 4 === 0) {
            n = Math.floor((n + 1) / 4) * 4 + 1;
        } else {
            n = Math.floor(n / 2);
        }
        count++;
    }
    return count;
};